src/platal_map.cpp

```diff
@@ -41,3 +38,17 @@ Object Map::RemoveObject(int obj_id){
     }
     return Object();
 }
+
+void Map::AddNpc(Character npc){
+    (*npc_)[npc.GetPosition()]= npc;
+}
+
+Character Map::RemoveNpc(int char_id){
+    for(std::map<Position*,Object>::iterator i = npc_->begin();i != npc_->end();i++){
+        if ((i->second).GetCharId()==char_id){
+            return i->second;
+        }
+    }
+    return Object();
+}
```

Comment:
You are returning an Object instead of a **Character**. The `std::map<Position*,**Object**>::iterator` is also faulty.


The bigger issue is that this and `RemoveObject` are not **removing** anything (it's the name of the method, the comment in the header is for additional clarification). [Use `map::erase()`](https://www.cplusplus.com/reference/map/map/erase/) instead of using iterator for nothing.